mixer/pkg/server/server.go

```diff
@@ -0,0 +1,292 @@
+	configStoreURL := a.ConfigStore2URL
+
+	if configStore2URL == "" {
+		configStore2URL = "k8s://"
+	}
+
+	if a.ServiceConfig != "" || a.GlobalConfig != "" {
+		s.configDir, err = serializeConfigs(a.GlobalConfig, a.ServiceConfig)
+		if err != nil {
+			_ = s.Close()
+			return nil, fmt.Errorf("Unable to serialize supplied configuration state: %v", err)
+		}
+		configStoreURL = "fs://" + s.configDir
+		configStore2URL = "fs://" + s.configDir
+	}
+
+	store2, err := store.NewRegistry2(config.Store2Inventory()...).NewStore2(configStore2URL)
+	if err != nil {
+		_ = s.Close()
+		return nil, fmt.Errorf("Failed to connect to the configuration server: %v", err)
+	}
+
+	var dispatcher mixerRuntime.Dispatcher
+	dispatcher, err = mixerRuntime.New(eval, evaluator.NewTypeChecker(), s.gp, s.adapterGP,
+		a.ConfigIdentityAttribute, a.ConfigDefaultNamespace,
+		store2, adapterMap, a.Templates)
+	if err != nil {
+		_ = s.Close()
+		return nil, fmt.Errorf("Failed to create runtime dispatcher: %v", err)
+	}
+
+	// Legacy Runtime
+	repo := template.NewRepository(a.Templates)
+	store, err := configStore(configStoreURL, a.ServiceConfigFile, a.GlobalConfigFile)
+	if err != nil {
+		_ = s.Close()
+		return nil, err
+	}
+
+	adapterMgr := adapterManager.NewManager(a.LegacyAdapters, aspect.Inventory(), evalForLegacy, s.gp, s.adapterGP)
+	configManager := config.NewManager(evalForLegacy, evaluator.NewTypeChecker(), adapterMgr.AspectValidatorFinder, adapterMgr.BuilderValidatorFinder,
+		a.Adapters,
+		adapterMgr.SupportedKinds,
+		repo, store, time.Second*time.Duration(a.ConfigFetchIntervalSec),
+		a.ConfigIdentityAttribute,
+		a.ConfigIdentityAttributeDomain)
+
+	configAPIServer := config.NewAPI("v1", a.ConfigAPIPort, evaluator.NewTypeChecker(),
+		adapterMgr.AspectValidatorFinder, adapterMgr.BuilderValidatorFinder, a.Adapters,
+		adapterMgr.SupportedKinds, store, repo)
+
+	configManager.Register(adapterMgr)
+	configManager.Register(ilEvalForLegacy)
+	configManager.Start()
+
+	go configAPIServer.Run()
+
+	// construct the gRPC options
+
+	var grpcOptions []grpc.ServerOption
+	grpcOptions = append(grpcOptions, grpc.MaxConcurrentStreams(uint32(a.MaxConcurrentStreams)))
+	grpcOptions = append(grpcOptions, grpc.MaxMsgSize(int(a.MaxMessageSize)))
+
+	var interceptors []grpc.UnaryServerInterceptor
+
+	var interceptor grpc.UnaryServerInterceptor
+
+	s.tracer, interceptor, err = startTracer(a.ZipkinURL, a.JaegerURL, a.LogTraceSpans)
+	if err != nil {
+		_ = s.Close()
+		return nil, fmt.Errorf("Unable to setup ZipKin: %v", err)
+	}
+	if interceptor != nil {
+		interceptors = append(interceptors, interceptor)
+	}
+
+	// setup server prometheus monitoring (as final interceptor in chain)
+	interceptors = append(interceptors, grpc_prometheus.UnaryServerInterceptor)
+	grpc_prometheus.EnableHandlingTimeHistogram()
+	grpcOptions = append(grpcOptions, grpc.UnaryInterceptor(grpc_middleware.ChainUnaryServer(interceptors...)))
+
+	if a.MonitoringPort != 0 {
```

Comment:
if we don't configure a monitoring port, should we not install the interceptors in the `grpcOptions`? in other words, should we support turning off monitoring completely by setting MonitoringPort == 0 ?

Reply:
I actually changed this behavior. You can now supply a monitoring port of 0 and it'll work fine. The system will pick a default port. This works much better for tests.